dapps/digital-twin-detail/src/components/sharing/SharingOverview.ts

```diff
@@ -0,0 +1,234 @@
+
+  sortBy = 'name';
+
+  filters = [
+    { type: 'unshared', icon: 'help-circle-outline' },
+    { type: 'user', icon: 'account-outline' },
+    { type: 'company', icon: 'office-building' },
+    { type: 'all', icon: 'share-variant' },
+  ];
+
+  /**
+   * Resolve permissions for the twin and containers.
+   */
+  async created(): Promise<void> {
+    this.columns = [
+      {
+        key: 'icon',
+        label: '',
+        thClass: 'th-icon',
+      },
+      {
+        key: 'name',
+        label: this.$t('_twin-detail.sharings.table.name'),
+        sortable: true,
+        tdClass: 'truncate',
+      },
+      {
+        key: 'containers',
+        label: this.$t('_twin-detail.sharings.table.containers'),
+        tdClass: 'truncate',
+      },
+    ];
+
+    await this.resolvePermissions();
+  }
+
+  /**
+   * Load all permissions for the current opened twins.
+   */
+  async resolvePermissions(): Promise<void> {
+    const runtime = this.getRuntime();
+    const { twin: { containerAddresses } }: { twin: DAppTwin } = this.$store.state;
+    const permittedUsers: { [address: string]: LoadingSharedUserInterface } = { };
+
+    // iterate through all containers and resolve the permissions
+    await Promise.all(containerAddresses.map(async (containerAddress: string): Promise<void> => {
+      /* Hint: Container API invites every shared user generally into the member role. As a result
+       * of this, we can simply resolve all members from the rights and roles, to check with which
+       * users anything was shared. */
+      const contract = runtime.contractLoader.loadContract('DataContract', containerAddress);
+      const roleMap = await runtime.rightsAndRoles.getMembers(contract);
+      const unique = Array.from(new Set([].concat(...Object.values(roleMap))));
+
+      // iterate through all permitted users and apply them to the permitted users mapping
+      unique.forEach((address: string): void => {
+        if (address !== runtime.activeAccount) {
+          if (!permittedUsers[address]) {
+            permittedUsers[address] = {
+              type: profileUtils.getProfileType(runtime, address),
+              name: profileUtils.getUserAliasFromAddress(runtime, address),
+              containers: [],
+            };
+          }
+
+          permittedUsers[address].containers.push(containerAddress);
+        }
+      });
+    }));
+
+    // iterate through all resolved profiles and push them into the datsa object
+    await Promise.all(Object.keys(permittedUsers).map(async (address: string): Promise<void> => {
+      this.data.push({
+        address,
+        containerNames: this.getContainerNames(permittedUsers[address].containers),
+        containers: permittedUsers[address].containers,
+        icon: profileUtils.getProfileTypeIcon(await permittedUsers[address].type),
+        name: await permittedUsers[address].name,
+        type: await permittedUsers[address].type,
+      });
+
+      this.sortHandler();
```

Comment:
would it make sense to sort after everything is finished?

Reply:
If we have a lot's of users, the users are added one after another to the list. For that, it makes sense to sort it any time, when a user gets pushed to the list. So the list will not hop afterwards.